pymongo/database.py

```diff
@@ -586,8 +616,10 @@ def collection_names(self, include_system_collections=True, session=None):
                 ReadPreference.PRIMARY) as (sock_info, slave_okay):
 
             wire_version = sock_info.max_wire_version
-            results = self._list_collections(
-                sock_info, slave_okay, session=session)
+
+            results = self._list_collections(sock_info,
+                                             slave_okay,
+                                             session=session)
```

Comment:
The ` if wire_version <= 2:` check on line 628 is unnecessary now that `_list_collections` strips the database name on MongoDB 2.6. Also, this should use the public `list_collections` method now.